src/GraphControl/Control/Grapher.cpp

```diff
@@ -433,6 +436,108 @@ namespace GraphControl
         }
     }
 
+    shared_ptr<IGraph> Grapher::GetGraph(GraphControl::Equation ^ equation)
+    {
+        std::shared_ptr<Graphing::IGraph> graph = m_solver->CreateGrapher();
+
+        wstringstream ss{};
+        ss << L"<math xmlns=\"http://www.w3.org/1998/Math/MathML\"><mrow><mi>show2d</mi><mfenced separators=\"\">";
+        ss << equation->GetRequest();
+        ss << L"</mfenced></mrow></math>";
+
+        wstring request = ss.str();
+        unique_ptr<IExpression> graphExpression;
+        if (graphExpression = m_solver->ParseInput(request))
+        {
+            if (graph->TryInitialize(graphExpression.get()))
+            {
+                return graph;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void Grapher::UpdateKeyGraphFeatures()
+    {
+        auto equations = GetValidEquations();
+        for (auto equation : GetValidEquations())
```

Comment:
Don't we only need to do this for the equation that had it's button pressed?

Reply:
Yes, I'll refactor it to only do analysis when the equation button is clicked after we refactor the equation class.